stmhal/stm32_it.c

```diff
@@ -347,15 +344,56 @@ void OTG_XX_WKUP_IRQHandler(void) {
     {}
 
     /* ungate PHY clock */
-     __HAL_PCD_UNGATE_PHYCLOCK((&pcd_handle));
+     __HAL_PCD_UNGATE_PHYCLOCK((&pcd_fs_handle));
   }
-#ifdef USE_USB_FS
+
   /* Clear EXTI pending Bit*/
   __HAL_USB_FS_EXTI_CLEAR_FLAG();
-#elif defined(USE_USB_HS)
+
+}
+#endif
+
+/**
+  * @brief  This function handles USB OTG HS Wakeup IRQ Handler.
+  * @param  None
+  * @retval None
+  */
+#if defined(USE_USB_HS)
+void OTG_HS_WKUP_IRQHandler(void) {
+
+  if ((&pcd_hs_handle)->Init.low_power_enable) {
+    /* Reset SLEEPDEEP bit of Cortex System Control Register */
+    SCB->SCR &= (uint32_t)~((uint32_t)(SCB_SCR_SLEEPDEEP_Msk | SCB_SCR_SLEEPONEXIT_Msk));
```

Comment:
Probably the code here that does the wake-up of the RCC that is common to the FS/HS can be split into a helper function to avoid code duplication.